src/main/kotlin/com/cognifide/gradle/aem/common/instance/service/osgi/OsgiFramework.kt

```diff
@@ -130,62 +174,187 @@ class OsgiFramework(sync: InstanceSync) : InstanceService(sync) {
         sync.http.post("$COMPONENTS_PATH/${component.id}", mapOf("action" to "disable"))
     }
 
+    /**
+     * Disable then enable again OSGi component. Works correctly even it is already disabled.
+     */
     fun restartComponent(pid: String) {
         disableComponent(pid)
         enableComponent(pid)
     }
 
-    fun findConfiguration(pid: String): Configuration? {
+    // ----- Configurations -----
+
+    /**
+     * Set properties for existing OSGi configuration.
+     */
+    fun configure(pid: String, properties: Map<String, Any?>) = updateConfiguration(pid, properties)
+
+    /**
+     * Get all OSGi configurations.
+     */
+    val configurations: Sequence<Configuration>
```

Comment:
@MatPiw  see that it is a sequence,
when using map then n HTTP requests are performed to AEM to load properties for all configurations which takes about 20 sec :) when having Sequence the consumer decides how process it, for instance in parallel, or break after some config found very quickly etc.